modules/tinymce/src/core/test/ts/browser/html/NodeTest.ts

```diff
@@ -394,4 +394,48 @@ describe('browser.tinymce.core.html.NodeTest', () => {
     node1 = root.append(new AstNode('b', 1));
     assert.isTrue(root.isEmpty({ img: 1 }, {}, isSpan), 'Should be true since the predicate says false.');
   });
+
+  context('children', () => {
+    it('TINY-7756: Returns empty array when there are no children', () => {
+      const root = new AstNode('p', 1);
+      assert.lengthOf(root.children(), 0);
+    });
+
+    it('TINY-7756: Returns single child', () => {
+      const root = new AstNode('p', 1);
+      const child = root.append(new AstNode('span', 1));
+
+      const children = root.children();
+      assert.lengthOf(children, 1);
+      assert.deepEqual(children[0], child);
```

Comment:
Should this and the below cases be `strictEquals` instead of `deepEquals`? As we aren't checking a nested object and instead want to make sure this is the same reference.